aws/flatcar-linux/kubernetes/workers/ami.tf

```diff
@@ -24,7 +23,7 @@ data "aws_ami" "coreos" {
 
   filter {
     name   = "name"
-    values = ["CoreOS-${local.flavor == "coreos" ? local.channel : "stable"}-*"]
+    values = ["CoreOS-${local.flavor == "coreos" ? local.channel : "stable"}-${local.flavor == "coreos" ? local.ver : ""}*"]
```

Comment:
I'm not sure about this logic here. Why is it needed? why can't we do `ver = var.os_version == "current" ? "*" : var.os_version` in `locals` and just use `local.ver` directly here?

Reply:
It's for the same reason we check the flavor before using the channel, and defaulting to stable: to avoid Terraform failing because no 'edge' AMI can be found in the 'coreos' images.

Both terraform "aws_ami" sections (flatcar and coreos) are always evaluated even if only one will end up being used.

When the user selects os_name `flatcar` and os_version `2345.99.1`, that version exists on AWS for 'flatcar' but not for 'coreos', and that would make terraform fail if we didn't guard it with this check.

Reply:
Ah, that make sense. We could possibly add `count` to both AMIs, to disable calculating the another one to simplify all that, but that seems like out of scope for this PR.

Thanks for explanation :)